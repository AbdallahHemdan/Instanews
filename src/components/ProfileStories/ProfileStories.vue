<template>
  <div class="profile-stories row">
    <div class="story" v-for="(story, index) in stories" :key="index">
      <img :src="story.storyImg" :alt="story.storyType" class="story__img" />
      <div class="story__title">{{ story.storyType }}</div>
    </div>
  </div>
</template>

<script>
export default {
  name: 'ProfileStories',
  data: function() {
    return {
      stories: [
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/71707560_143278756997187_4954273971273726619_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=9enQ_--0NiEAX-PrSWg&oh=b52ae838299ad24b24f8a32084d72dd1&oe=5F505948',
          storyType: 'Tunes',
        },
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/73372120_205078993821310_2769530515492845343_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=fMGWzkMucqsAX8IAWVS&oh=e167fec9c92851284e90f72ab5bcb98a&oe=5F512C02',
          storyType: 'Tech Things',
        },
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/72486838_139345537366386_2211609228713571660_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=Jnmt_xcKNzgAX92U1TU&oh=3410e7b4e1da0e574da43caade4fa510&oe=5F51069F',
          storyType: 'Pets and Pats',
        },
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/91987627_202824527805182_6378536052041441788_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=rOqREZLVg8AAX_QB0aG&oh=e17870c95ae8d27671722bd75e610ead&oe=5F510CFC',
          storyType: 'Fooood',
        },
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/75443204_150603682858933_1559794405085010385_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=mPy0JMY0LrAAX-TGVWI&oh=3e8ca1c0029a48c82f77130f5e1e1487&oe=5F4F88FF',
          storyType: 'OOTD',
        },
        {
          storyImg:
            'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/72760401_1180474338818724_660695432915614834_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=wA7pQnteh8QAX-26pWb&oh=ec8844788df43cf6cb2de273898df945&oe=5F51A957',
          storyType: 'Slice of Life',
        },
      ],
    };
  },
};
</script>

<style lang="scss" scoped>
.profile-stories {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: center;
  margin-top: 2rem;
  margin-bottom: 1rem;

  .story {
    display: flex;
    align-items: center;
    flex-direction: column;
    margin-right: 1rem;
    margin-bottom: 1rem;
    cursor: pointer;

    @media (max-width: 576px) {
      margin-left: 0;
    }

    &__img {
      width: 80px;
      height: 80px;
      border-radius: 50%;
      background-color: transparent;
      border: 2px solid $light-gray;
      margin-bottom: 0.5rem;
      padding: 2px;
    }

    &__title {
      color: $main-color;
      font-weight: 400;
    }
  }
}
</style>
