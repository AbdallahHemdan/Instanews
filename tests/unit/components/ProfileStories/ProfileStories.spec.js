import { shallowMount } from '@vue/test-utils';
import ProfileStories from './../../../../src/components/ProfileStories/ProfileStories.vue';

describe('ProfileStories.vue', () => {
  let cmp;
  const stories = [
    {
      storyImg:
        'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/71707560_143278756997187_4954273971273726619_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=9enQ_--0NiEAX-PrSWg&oh=b52ae838299ad24b24f8a32084d72dd1&oe=5F505948',
      storyType: 'Tunes',
    },
    {
      storyImg:
        'https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/73372120_205078993821310_2769530515492845343_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=fMGWzkMucqsAX8IAWVS&oh=e167fec9c92851284e90f72ab5bcb98a&oe=5F512C02',
      storyType: 'Tech Things',
    },
  ];

  beforeEach(() => {
    cmp = shallowMount(ProfileStories, {
      data: function() {
        return {
          stories,
        };
      },
    });
  });

  describe('Testing render of data', () => {
    it('should render all the stories in correctly', () => {
      expect(cmp.findAll('.story').length).toBe(stories.length);
    });

    it('should render stories images data correctly', () => {
      expect(
        cmp
          .findAll('.story')
          .at(0)
          .find('.story__img')
          .attributes('src'),
      ).toBe(stories[0].storyImg);
    });

    it('should render stories titles data correctly', () => {
      expect(
        cmp
          .findAll('.story')
          .at(0)
          .find('.story__title')
          .text(),
      ).toBe(stories[0].storyType);
    });
  });

  describe('Snapshot testing', () => {
    it('should pass snapshot testing', () => {
      expect(cmp.element).toMatchSnapshot();
    });
  });
});
