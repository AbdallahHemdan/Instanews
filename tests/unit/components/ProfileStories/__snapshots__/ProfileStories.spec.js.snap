// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileStories.vue Snapshot testing should pass snapshot testing 1`] = `
<div
  class="profile-stories row"
>
  <div
    class="story"
  >
    <img
      alt="Tunes"
      class="story__img"
      src="https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/71707560_143278756997187_4954273971273726619_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=9enQ_--0NiEAX-PrSWg&oh=b52ae838299ad24b24f8a32084d72dd1&oe=5F505948"
    />
     
    <div
      class="story__title"
    >
      Tunes
    </div>
  </div>
  <div
    class="story"
  >
    <img
      alt="Tech Things"
      class="story__img"
      src="https://instagram.fcai19-2.fna.fbcdn.net/v/t51.2885-15/s150x150/73372120_205078993821310_2769530515492845343_n.jpg?_nc_ht=instagram.fcai19-2.fna.fbcdn.net&_nc_ohc=fMGWzkMucqsAX8IAWVS&oh=e167fec9c92851284e90f72ab5bcb98a&oe=5F512C02"
    />
     
    <div
      class="story__title"
    >
      Tech Things
    </div>
  </div>
</div>
`;
